samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java

```diff
@@ -1,2 +1,133 @@
+            languagePreference.set(turnContext, lang);
+            Activity reply = MessageFactory.text(String.format("Your current language code is: %1$s", lang));
+            turnContext.sendActivity(reply);
+            return userState.saveChanges(turnContext, false);
+        } else {
+            // Show the user the possible options for language. If the user chooses a different language
+            // than the default, then the translation middleware will pick it up from the user state and
+            // translate messages both ways, i.e. user to bot and bot to user.
+            Activity reply = MessageFactory.text("Choose your language:");
+            CardAction esAction = new CardAction();
+            esAction.setTitle("Español");
+            esAction.setType(ActionTypes.POST_BACK);
+            esAction.setValue(ENGLISH_SPANISH);
+            CardAction enAction = new CardAction();
+            enAction.setTitle("English");
+            enAction.setType(ActionTypes.POST_BACK);
+            enAction.setValue(ENGLISH_ENGLISH);
+            List<CardAction> actions = new ArrayList<>(Arrays.asList(esAction, enAction));
+            SuggestedActions suggestedActions = new SuggestedActions();
+            suggestedActions.setActions(actions);
+            return turnContext.sendActivity(reply).thenApply(resourceResponse -> null);
+        }
+    }
+
+    private static CompletableFuture<Void> sendWelcomeMessage(TurnContext turnContext) {
+        for (ChannelAccount member: turnContext.getActivity().getMembersAdded()) {
+            // Greet anyone that was not the target (recipient) of this message.
+            // To learn more about Adaptive Cards, see https://aka.ms/msbot-adaptivecards for more details.
+            if (!member.getId().equals(turnContext.getActivity().getRecipient().getId())) {
+                Attachment welcomeCard = createAdaptiveCardAttachment();
+                Activity response = MessageFactory.attachment(welcomeCard);
+                return turnContext.sendActivity(response)
+                    .thenCompose(task -> turnContext.sendActivity(MessageFactory.text(WELCOME_TEXT))
+                    .thenApply(resourceResponse -> null));
+            }
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    private static Attachment createAdaptiveCardAttachment() {
+        // combine path for cross platform support
+        try {
+            Path path = Paths.get(".", "Cards", "welcomeCard.json");
+            BufferedReader reader = Files.newBufferedReader(path);
+            String adaptiveCard = reader.readLine();
+            Attachment attachment = new Attachment();
+            attachment.setContentType("application/vnd.microsoft.card.adaptive");
+            ObjectMapper mapper = new ObjectMapper();
+            attachment.setContent(mapper.readValue(adaptiveCard, String.class));
```

Comment:
This can be reduced following [this](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-integration-core/src/main/java/com/microsoft/bot/integration/AdapterWithErrorHandler.java#L97), also the return of the object